lib/marked.js

```diff
@@ -146,6 +146,12 @@ Lexer.prototype.lex = function(src) {
  * Lexing
  */
 
+Lexer.prototype.id_generator = function(title, explicit_id) {
+	var draft = explicit_id ? explicit_id : title;
+	draft = this.options.headerPrefix + draft;
+	return draft.toLowerCase().replace(/[^\w]+/g, '-');
```

Comment:
Explicit ID should be returned as is, without being case-converted or transformed in any way. If author supplied explicit IDs, there must be a good reason why he did so. Hence it should be used as is. No point in transforming it any further.